crates/wdl-cli/src/inputs/file.rs

```diff
@@ -111,31 +201,121 @@ impl InputFile {
+        let err = InputFile::read(&"./tests/fixtures/nonmap_inputs.json".parse().unwrap())
+            .await
+            .unwrap_err();
+        assert_eq!(
+            err.to_string().replace("\\", "/"),
+            "input file `tests/fixtures/nonmap_inputs.json` did not contain a map from strings to \
+             values at the root"
+        );
 
         // A YML file that does not have a map at the root.
-        let err = InputFile::read(Path::new("./tests/fixtures/nonmap_inputs.yml")).unwrap_err();
-        assert!(matches!(
-            err,
-            Error::NonMapRoot(path) if path.to_str().unwrap() == "./tests/fixtures/nonmap_inputs.yml"
-        ));
+        let err = InputFile::read(&"./tests/fixtures/nonmap_inputs.yml".parse().unwrap())
+            .await
+            .unwrap_err();
+        assert_eq!(
+            err.to_string().replace("\\", "/"),
+            "input file `tests/fixtures/nonmap_inputs.yml` did not contain a map from strings to \
+             values at the root"
+        );
+    }
+
+    #[tokio::test]
+    async fn missing_ext() {
+        let err = InputFile::read(&"./tests/fixtures/missing_ext".parse().unwrap())
+            .await
+            .unwrap_err();
+        assert_eq!(
+            err.to_string().replace("\\", "/"),
+            "unsupported input file `tests/fixtures/missing_ext`: the supported formats are JSON \
+             (`.json`) or YAML (`.yaml` and `.yml`)"
+        );
+
+        let err = InputFile::read(&"http://example.com".parse().unwrap())
+            .await
+            .unwrap_err();
+        assert_eq!(
+            err.to_string(),
+            "unsupported input file `http://example.com/`: the supported formats are JSON \
+             (`.json`) or YAML (`.yaml` and `.yml`)"
+        );
+    }
+
+    #[tokio::test]
+    async fn read_local() {
+        let inputs = InputFile::read(&"./tests/fixtures/inputs_one.json".parse().unwrap())
+            .await
+            .unwrap();
+
+        let inner = inputs.into_inner();
+        assert_eq!(inner.len(), 3);
+
+        let expected_origin = absolute(Path::new("tests/fixtures")).unwrap();
+        let expected_origin = expected_origin.to_str().unwrap();
+
+        let (origin, value) = &inner["foo"];
+        assert_eq!(origin.to_str().unwrap(), expected_origin);
+        assert_eq!(value.as_str().unwrap(), "bar");
+
+        let (origin, value) = &inner["baz"];
+        assert_eq!(origin.to_str().unwrap(), expected_origin);
+        assert_eq!(value.as_number().unwrap().as_f64().unwrap() as u64, 42);
+
+        let (origin, value) = &inner["quux"];
+        assert_eq!(origin.to_str().unwrap(), expected_origin);
+        assert_eq!(value.as_str().unwrap(), "qil");
+    }
+
+    #[tokio::test]
+    async fn read_remote() {
+        let inputs = InputFile::read(&"https://gist.githubusercontent.com/peterhuene/9990b86bf0c419e144326b0276bf6f14/raw/d4116ef8888ccd78e2967d7ad32e1aeb3e4ab734/inputs.json".parse().unwrap())
```

Comment:
is this intentional?

Reply:
👍  seems fine but also seems like a potentially accidental commit so wanted to check